### PR TITLE
feat: Add navigation bar style to product details 🎃

### PR DESCRIPTION
**Pull request by oradwanomar**

### CODE REVIEW

File: Dokan/Dokan/Scenes/ProductDetails/Views/ProductDetailsNavigationBarBehavior.swift [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    private unowned var navigationItem: UINavigationItem
+    private var onRedo: () -> () = {}
+    private var onCart: () -> () = {}
+    private var navigationBarButtonItems: [UIBarButtonItem] = []
+    
+    init(navigationItem: UINavigationItem) {
+        self.navigationItem = navigationItem
+    }
+    
+    func configure(onRedo: @escaping () -> () ,onCart: @escaping () -> ()) {
```
Comment (ahmdyasser) [inline Dokan/Dokan/Scenes/ProductDetails/Views/ProductDetailsNavigationBarBehavior.swift]:
```suggestion
    func configure(onRedo: @escaping () -> (), onCart: @escaping () -> ()) {
```

File: Dokan/Dokan/Scenes/ProductDetails/Views/ProductDetailsNavigationBarBehavior.swift [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        
+        let redoButton = UIBarButtonItem(image: UIImage(named: "redo")?.withRenderingMode(.alwaysOriginal),
+                                     style: .done,
+                                     target: self,
+                                     action: #selector(redoWasTapped))
+        let cartButton = UIBarButtonItem(image: UIImage(named: "shopping-cart")?.withRenderingMode(.alwaysOriginal),
+                                         style: .done,
+                                         target: self,
+                                         action: #selector(cartWasTapped))
+        navigationBarButtonItems = [cartButton,redoButton]
```
Comment (ahmdyasser) [inline Dokan/Dokan/Scenes/ProductDetails/Views/ProductDetailsNavigationBarBehavior.swift]:
```suggestion
        navigationBarButtonItems = [cartButton, redoButton]
```

File: Dokan/Dokan/Scenes/ProductDetails/Views/ProductDetailsNavigationBarBehavior.swift [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        navigationItem.rightBarButtonItems = navigationBarButtonItems
+        navigationItem.backBarButtonItem = UIBarButtonItem(title: "", style: .done, target: self, action: nil)
+    }
+    
+    func updateCartButton() {
+        let cartbadgeButton = UIBarButtonItem(image: UIImage(named: "shopping-cart-badge")?.withRenderingMode(.alwaysOriginal),
```
Comment (ahmdyasser) [inline Dokan/Dokan/Scenes/ProductDetails/Views/ProductDetailsNavigationBarBehavior.swift]:
```suggestion
        let cartBadgeButton = UIBarButtonItem(image: UIImage(named: "shopping-cart-badge")?.withRenderingMode(.alwaysOriginal),
```

**Review by ahmdyasser — COMMENTED** [review]
Linting changes

**Review by ahmdmhasn — APPROVED** [review]
🚀

**Review by oradwanomar — COMMENTED** [review]
Approve ya Yasser @ahmdyasser

**Review by ahmdyasser — APPROVED** [review]

**Pull request merged**